harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -1235,6 +1236,20 @@ def test_trial_validation_checkpointing(self, tmp_path: pathlib.Path):
             controller.core_context.train.get_experiment_best_validation.reset_mock()
             controller._checkpoint.reset_mock()
 
+    @mock.patch.object(det.core.DummySearcherOperation, "report_progress")
+    def test_searcher_progress_reporting(self, mock_report_progress: mock.MagicMock):
+        trial, controller = pytorch_utils.create_trial_and_trial_controller(
+            trial_class=pytorch_onevar_model.OneVarTrial,
+            scheduling_unit=10,
+            hparams=self.hparams,
+            trial_seed=self.trial_seed,
+            max_batches=100,
+        )
+        controller.run()
+        assert mock_report_progress.call_count == 100 // 10
+        for i, step in enumerate(range(10, 110, 10)):
+            assert mock_report_progress.call_args_list[i].args[0] == step
```

Comment:
I think this could be more expressive, and easier to understand in error cases, if you wrote it like this:

```
exp_prog = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
got_prog = [x.args[0] for x in mock_report_progress.call_args_list]
assert exp_prog == got_prog
```

Then it's way easier to see what's expected, and in failure cases you get to see the whole list of exp vs got, rather than just a point failure like (`assert failed: 30 != 40`)